### ISSUE TITLE
Support for `toNotHaveStyleRule`

### ISSUE THREAD

**Issue by jacopotarantino** [mbasso/styled-components-test-utils#9]
**General Information**

- [ ] Bug
- [x] Improvement
- [ ] Feature
- [ ] Other

**Description**

Similar to how other utilities provide negative matchers (such as expect's [toNotExist](https://github.com/mjackson/expect#tonotexist)) I would like to either see `toNotHaveStyleRule` supported in `styled-components-test-utils` or would love advice on how to achieve the same effect without updating this library. I came looking for this feature because I would like to write a test case for a styled component where I assert that a style is *not* present and then in a second test case, with a prop set, assert that the style *is* present. That way I have full coverage of both possible branches. Please let me know if any of this is not clear. Thanks in advance for the help!

**Versions**

- styled-components-test-utils: 0.4.1
- node: 8.11.4

- jest: n/a
- expect: 1.20.2
- chai: n/a
- jasmine: n/a

**Comment by mbasso** [comment]
Hi @jacopotarantino,
thank you for opening this issue.
I think that this is a great improvement, so I've just created a new branch with this feature. At the moment we can use it in this way:

> expect(tree).toNotHaveStyleRule(selector)
>
> expect({ component, modifier, media }).toNotHaveStyleRule(selector)
>
> expect({ css, props, modifier, media }).toNotHaveStyleRule(selector)

for example:

```js
const Button = styled.button`
  color: blue;
`;
const component = renderer.create(<Button />);
expect(component).toNotHaveStyleRule('background-color');
```

What do you think about it? Does it look good?

**Comment by alansouzati** [comment]
since this is a generic utility I think this is a good solution.

In chai I'm doing `expect(component).not.toHaveStyleRule('background-color');` and this is working fine.

**Comment by mbasso** [comment]
@jacopotarantino @alansouzati 
Just published [styled-components-test-utils@0.5.0](https://github.com/mbasso/styled-components-test-utils/releases/tag/0.5.0) 🎉
I'll close this issue, don't hesitate to reopen it if necessary. Thank you! 😄

**Comment by jacopotarantino** [comment]
@mbasso i know i'm slow on this one but thank you so much for the update!!! i really appreciate it.
